### PR TITLE
Add command to create appointments

### LINKED ISSUE #26: As a user, I can schedule an appointment for the patient.

### PR DESCRIPTION
**Pull request by dayn-1**
Add support for creating new appointments.
Newly added modules relating to appointments have yet to have test cases written.

Closes #26.

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/date/Date.java [hunk 1]
```diff
@@ -1,34 +1,56 @@
...
+     * Returns if date is before today
+     */
+    public static boolean isBeforeToday(String date) {
+        return LocalDate.parse(date, DATE_FORMATTER).isBefore(LocalDate.now());
     }
 
     @Override
     public String toString() {
-        return this.value;
+        return value.format(DATE_FORMATTER);
```
Comment (limrik) [inline src/main/java/seedu/address/commons/core/date/Date.java]: Just a nit but we should return `this.value.format(DATE_FORMATTER)` as value is a class attribute

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 2]
```diff
@@ -13,4 +13,8 @@ public class CliSyntax {
     public static final Prefix PREFIX_EMAIL = new Prefix("e/");
     public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
     public static final Prefix PREFIX_TAG = new Prefix("t/");
+    public static final Prefix PREFIX_DATE = new Prefix("d/");
+    public static final Prefix PREFIX_START_TIME = new Prefix("fr/");
+    public static final Prefix PREFIX_END_TIME = new Prefix("to/");
+    public static final Prefix PREFIX_NOTE = new Prefix("nt/");
```
Comment (limrik) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java]: Shall we consider using the prefixes date/ from/ and note/ instead? I feel like the current shorter form of prefixes may obscure the meaning and make it harder for users to use.

**Review by limrik — APPROVED** [review]
Just a few minor comments, if not LGTM! Thanks @dayn-1 !

File: src/main/java/seedu/address/logic/commands/AddAppCommand.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            + PREFIX_NOTE + "NOTE ...\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NRIC + "T0123456A "
+            + PREFIX_DATE + "2024-02-20 "
+            + PREFIX_START_TIME + "11:00 "
+            + PREFIX_END_TIME + "11:30 "
+            + PREFIX_TAG + "Medical Check-up "
+            + PREFIX_NOTE + "May come earlier ";
+
+    public static final String MESSAGE_SUCCESS = "New appointment added: %1$s";
```
Comment (ThatGirlJam) [inline src/main/java/seedu/address/logic/commands/AddAppCommand.java]: Just a nit, perhaps the Message success can be specified to be MESSAGE_ADD_APPOINTMENT_SUCCESS ?

File: src/main/java/seedu/address/logic/commands/AddAppCommand.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    public static final String MESSAGE_DUPLICATE_APPOINTMENT = "This appointment already exists in CLInic";
```
Comment (ThatGirlJam) [inline src/main/java/seedu/address/logic/commands/AddAppCommand.java:44]: Should the error messages like MESSAGE_DUPLICATE_APPOINTMENT and MESSAGE_PATIENT_NOT_FOUND be extracted to appear in Messages instead? Perhaps they could be reused in the CancelAppCommand class later on?

**Review by ThatGirlJam — COMMENTED** [review]
Hi Dayn! Thanks for the code, just left a few comments for consideration 👍

File: src/main/java/seedu/address/model/Model.java [hunk 5]
```diff
@@ -84,4 +98,38 @@ public interface Model {
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredPersonList(Predicate<Person> predicate);
+
+    /**
+     * Returns true if an appointment with the same identity as {@code appointment} exists in CLInic.
+     */
+    boolean hasAppointment(Appointment appointment);
```
Comment (xuelinglow) [inline src/main/java/seedu/address/model/Model.java:105]: Maybe can write hasAnotherAppointment instead to make it clearer that it returns true if duplicate

**Review by xuelinglow — COMMENTED** [review]
Overall looks good!

**Pull request merged**